libs/sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -124,8 +124,8 @@ public static String replaceEscapedChars(String input) {
                         i = endIdx - 1; // skip already consumed chars
                         break;
                     default:
-                        // non-valid escaped char sequence
-                        builder.append(currentChar);
+                        // Apply rule of the C-Style escaped sequence: any other character following a backslash is taken literally.
```

Comment:
I'm a bit confused with this comment vs the behavior of tests. If we have `E'<string>'` then `E'\t'` becomes `t`, right? If we have `E'\\t'` it becomes `t` again, but in psql:
```
matriv=> select E'Te\\%tch';
 ?column? 
----------
 Te\%tch
(1 row)
```

Reply:
tested on this PR:
```
select E'\t'; 
+--+
| '     '  |
+--+
|  |
+--+
SELECT 1 row in set (0.002 sec)
```

Reply:
BTW, this probably indicates that commit message/change entry is poorly phrased. Let me know if you have an idea how to describe this better

Reply:
thx for explaining!

Reply:
Again, unless I'm still confused, the title should be: `Ignore leading backslash escaping a backslash + char sequence` ?

Reply:
Thanks for the suggestion, I would use it with some adaptations:
1. I think we need to mention C-Style escaped string since it's only about those.
2. It's about backslash + not special char

What about 
"Ignore leading backslash if it doesn't resolve to a special character in combination with the next character"?

I will use it for commit message - but for change entry it's too much details so I keep current (+example)?

Reply:
Actually, easier to say 
"Ignore backslash in C-Style escaped string unless it's not escaped by another backslash."

Reply:
loos like had grammar issue" should be  unless it's escaped (I dropped "not" in the final version)